src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -171,11 +170,29 @@ export class AddScenarioModal implements OnInit, OnDestroy {
 
     addError() {
         const control = <FormArray>this.newScenForm.get('regressionRegions.regressions.errors');
-        control.push(new FormControl(null, Validators.required));
+        control.push(this._fb.group({
+            id: new FormControl(null, Validators.required),
+            value: new FormControl(null, Validators.required)
+        }));
+    }  
+
+    fillError(id,index){
+        const errorList = <FormArray>this.newScenForm.get('regressionRegions.regressions.errors');
+        const error = errorList.at(index).value;
+        const selectedError = this.errors.find(x => x.id == id);
+        error.name = selectedError.name;
+        error.code = selectedError.code;
     }
 
     showMathjax() {
         const exp = this.newScenForm.get('regressionRegions.regressions.equation').value;
+        if(exp == null){
+            this.equation = " ";
+            const equ = document.getElementById('mathjaxEq');
+            equ.style.visibility = 'hidden';
+            if (equ.firstChild) {equ.removeChild(equ.firstChild); }
+            equ.insertAdjacentHTML('afterbegin', '<span [MathJax]="equation">' + this.equation + '</span');
+        }else{
```

Comment:
Can you walk me through the changes you made to this section? It looks like there's some repetition with the else below, and I'm wondering if that can be avoided.

Reply:
![image](https://user-images.githubusercontent.com/59063181/72628164-64693600-3913-11ea-9cdf-bc857afed0ea.png)

I was getting a weird null character when I cleared the form so I was trying to fix that. Your right though, there was a lot of repetition. I tried to rewrite it better in the last commit.

Reply:
Ah, ok. Thanks for explaining and for cleaning that up!